attr.c

```diff
@@ -215,7 +227,30 @@ int iio_add_attr(union iio_pointer p, struct iio_attr_list *attrs,
 
 	attrs->num++;
 
-	iio_sort_attrs(attrs);
+	if (lvalues) {
+		attrs_and_vals.attr_val_pairs = malloc(attrs->num * sizeof(struct iio_attr_val));
```

Comment:
why not just pass `lvalues` to `iio_sort_attrs`  and then if `lvalues` isn't null, combine/separate them there - it will keep everything a little cleaner (I think).

Reply:
It might help a bit, but I would go back to the first implementation where I was inserting the value of the attribute.
I kept the first commit (that's why I reverted it so it can still be seen here in this PR) to compare its complexity with this approach. The 1st is simpler and more efficient. In the 2nd I am doing quite a few of `mallocs` and `memcpys`.
The thing is that these context attributes are the exception as they are the only ones for which we need to keep the list of their corresponding values. For all the other attributes (device, channel, buffer, debug), the function `iio_add_attr()` fits perfectly.

Reply:
maybe the "right" answer is not to use qsort, and just a simple sort algorithm that you do yourself, that can manage both structures... (to keep them sync'ed) That might be the simple answer...

Reply:
plus - I think we need this when iio_emu shows up (doesn't it?) - so this would be useful.

Reply:
I am keeping the initial implementation for now as I can't anticipate the exact form in which the attributes will be kept with iio_emu. But I would use your idea to have a way to sort both attributes and values when the time comes.